eng/pipelines/templates/steps/sync-repo-branch.yml

```diff
@@ -1,17 +1,102 @@
 parameters:
-  Repos: # Provde a list of repos
+  # Provde a list of repos
+  # Example:
+  #   Azure/azure-rest-api-specs:
+  #     Branch: master
+  #     TargetRepos: 
+  #       Azure/azure-rest-api-specs-pr:
+  #       azure-sdk/azure-rest-api-specs:
+  #         Branch: dev
+  #         Rebase: true
+  #   Azure/azure-sdk-for-go:
+  #     Branch: master
+  #     TargetRepos: 
+  #       Azure/azure-sdk-for-go-pr:
+  - name: Repos
+    type: object
+  # Github PAT
+  - name: GH_TOKEN
+    type: string
+  # Github alias for git config
+  - name: GithubAlias
+    type: string
+    default: azure-sdk
+
 
 steps:
 
+- pwsh: |
+    git config --global user.email "${{ parameters.GithubAlias }}@microsoft.com"
+    git config --global user.name "${{ parameters.GithubAlias }}"
+  displayName: Init git config
+
 - ${{ each repo in parameters.Repos }}:
-  - script: |
-      git clone --filter=blob:none --branch ${{ repo.value.Branch }} https://github.com/${{ repo.key }} ${{ repo.key }}
+  - pwsh: |
+      Set-PsDebug -Trace 1
```

Comment:
Why do we need all this extra logic. I agree we need to pass the GH_TOKEN to the clone but beyond that I believe everything else is already handled by the clone itself.

Reply:
Resolved

Reply:
We may enter this with same repo.key multiple times. A simple clone would not work here.

Reply:
Why would we ever need to enter more than once?

Reply:
OK I think I can understand why it might be interesting to potentially allow multiple calls to this per-repo. If you need to mirror more then one branch would be an example.